oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/userinfo/DefaultOAuth2UserServiceTests.java

```diff
@@ -158,6 +158,40 @@ public void loadUserWhenUserInfoSuccessResponseThenReturnUser() {
 		assertThat(userAuthority.getAttributes()).isEqualTo(user.getAttributes());
 	}
 
+	@Test
+	public void loadUserWhenNestedUserInfoSuccessThenReturnUser() {
+		// @formatter:off
+		String userInfoResponse = "{\n"
+				+ "   \"user\": {\"user-name\": \"user1\"},\n"
```

Comment:
I believe the reason you are having trouble with the simple dot syntax is since `user-name` is not a valid SpEL accessor. 

Either change the payload to use `username` for the property or use `user['user-name']` for the expression.